internal/graph/graph.go

```diff
@@ -213,3 +214,47 @@ func (g *Graph[V]) TopologicallySortWithPriority(isLowerPriority func(V, V) bool
 
 	return output, nil
 }
+
+func (g *Graph[V]) EncodeDOT(w io.Writer, sortVertices bool) (err error) {
+	builder := &dotBuilder{w}
+	err = builder.init()
+	if err != nil {
+		return err
+	}
+	defer func() {
+		err = builder.finish()
```

Comment:
Deferrals are great for closing resources that we shouldn't leave open in the event of an error, e.g., db connections, files. 

In this case, it's okay if we don't "finish()" the builder even if an error occurs. So to simplify the code a bit, just call `builder.finish()` at the end of the function rather than deferring it